internal/blocksync/reactor.go

```diff
@@ -186,15 +186,21 @@ func (r *Reactor) OnStop() {
 func (r *Reactor) respondToPeer(ctx context.Context, msg *bcproto.BlockRequest, peerID types.NodeID, blockSyncCh *p2p.Channel) error {
 	block := r.store.LoadBlock(msg.Height)
 	if block != nil {
+		extCommit := r.store.LoadBlockExtendedCommit(msg.Height)
+		if extCommit == nil {
+			return fmt.Errorf("blockstore has block but not extended commit (block: %v)", block)
+		}
```

Comment:
Entirely optional readability nit:
```suggestion
			return fmt.Errorf("found block in store without extended commit: %v", block)
		}
```